File: go-sdk/pkg/models/operations/updateabranchpassword.go
package operations

type UpdateABranchPasswordPathParams struct {
	Branch       string `pathParam:"style=simple,explode=false,name=branch"`
	Database     string `pathParam:"style=simple,explode=false,name=database"`
	ID           string `pathParam:"style=simple,explode=false,name=id"`
	Organization string `pathParam:"style=simple,explode=false,name=organization"`
}

type UpdateABranchPasswordRequestBody struct {
	DisplayName      string  `json:"display_name"`
	ReadOnlyRegionID *string `json:"read_only_region_id,omitempty"`
}

type UpdateABranchPasswordRequest struct {
	PathParams UpdateABranchPasswordPathParams
	Request    *UpdateABranchPasswordRequestBody `request:"mediaType=application/json"`
}

type UpdateABranchPasswordResponse struct {
	ContentType                                   string
	StatusCode                                    int64
	UpdateABranchPassword200ApplicationJSONObject map[string]map[string]interface{}
}

File: ts-sdk/src/sdk/models/operations/updateabranchpassword.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";


export class UpdateABranchPasswordPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=branch" })
  branch: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=database" })
  database: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=organization" })
  organization: string;
}

export class UpdateABranchPasswordRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=display_name" })
  displayName: string;

  @SpeakeasyMetadata({ data: "json, name=read_only_region_id" })
  readOnlyRegionId?: string;
}

export class UpdateABranchPasswordRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: UpdateABranchPasswordPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: UpdateABranchPasswordRequestBody;
}

export class UpdateABranchPasswordResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  updateABranchPassword200ApplicationJSONObject?: Record<string, Record<string, any>>;
}